backend/infrahub/permissions/report.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from infrahub.core.account import fetch_permissions
+from infrahub.core.constants import GlobalPermissions, PermissionDecision
+from infrahub.core.registry import registry
+from infrahub.permissions.local_backend import LocalPermissionBackend
+
+if TYPE_CHECKING:
+    from infrahub.core.schema import MainSchemaTypes
+    from infrahub.database import InfrahubDatabase
+    from infrahub.graphql import GraphqlContext
+    from infrahub.permissions.types import KindPermissions
+
+
+async def report_schema_permissions(
+    db: InfrahubDatabase, schemas: list[MainSchemaTypes], context: GraphqlContext
+) -> list[KindPermissions]:
+    permissions = await fetch_permissions(
+        account_id=context.active_account_session.account_id, db=db, branch=context.branch
+    )
+
+    restrict_changes = False
+    if context.branch.name == registry.default_branch:
+        restrict_changes = True
+        for global_permission in permissions["global_permissions"]:
+            if global_permission.action == GlobalPermissions.EDIT_DEFAULT_BRANCH.value:
+                restrict_changes = False
+
+    checker = LocalPermissionBackend()
```

Comment:
I know we aren't fully committed to dependency injection, but if you want to use dependency injection here, then `report_schema_permissions` could become a method that is something like  `SchemaPermissionsReporter.report` and `SchemaPermissionsReporter` accepts a `PermissionBackend` in its constructor.

something like
```
class SchemaPermissionsReporter:
    def __init__(self, checker: PermissionBackend):
        self.checker = checker

    async def report(...) -> list[KindPermssion]:
       ...
```

Reply:
I think it makes sense to explore this when we're adding more checks later on.